src/pepr/operator/controllers/keycloak/client-sync.ts

```diff
@@ -84,7 +84,7 @@ export async function purgeSSOClients(pkg: UDSPackage, newClients: string[] = []
     const token = Store.getItem(storeKey);
     if (token) {
       await apiCall({ clientId: ref }, "DELETE", token);
-      Store.removeItem(storeKey);
```

Comment:
Why this change? We are already awaited the DELETE to keycloak.  

I think waiting for the remove only makes sense if the potential update (if someone deleting and created the same client?) would result in some sort of conflict?

Reply:
This is a bit of a nit: because I think ultimately it doesn't matter but I think its ultimately a question of if this works as expected?

```
Store.setItem("foo". "bar)
Store.removeItem("foo")
```

Reply:
If the token still existed in the store on a subsequent creation we would try to use it ([ref](https://github.com/defenseunicorns/uds-core/blob/finalizer-add/src/pepr/operator/controllers/keycloak/client-sync.ts#L139)) on the first attempt, and then log an error + retry without the token on a second try. It would mostly just be a bit of clutter in the logs if we didn't wait for the token deletion OR best case we are gambling on the timing between token actually being removed and "new" client being requested.

Imo the whole point of the finalizer and this cleanup is to guarantee that everything connected to the Package is truly gone when it is deleted from the cluster. I'm not sure why we would do that with some pieces but not everything, but I might just be missing your point here with the token?

Reply:
After chatting about this you are correct.  I had forgotten about the store being used as an indication of whether or not the client existed.  

As a follow up to our conversation about if `removeItem` optimistically removed items from the local cache.
```
Store.removeItem("foo")
Store.getItem("foo")
```
It seems like you are right there too.  It does not, which means things go like this: 
```
# waits up to 5 seconds to flush/write this removal to kubernetes
Store.removeItem("foo")

# gets item from in memory cache
# in memory cache is updated exclusively from a 5 second debounced watch 
Store.getItem("foo")
```